File: frontend/erpclink-web/src/app/features/procurement/models/purchase-order.models.ts
export interface PurchaseOrderLineDto {
  id: string;
  catalogItemId?: string | null;
  descriptionSnapshot: string;
  quantity: number;
  unitCost: number;
  lineDiscountAmount: number;
  lineSubtotal: number;
  lineTotal: number;
  sortOrder: number;
}

export interface PurchaseOrderDto {
  id: string;
  organizationId: string;
  branchId: string;
  purchaseOrderNumber: string;
  supplierId: string;
  orderDate: string;
  status: string;
  currencyCode: string;
  subTotal: number;
  discountAmount: number;
  taxAmount: number;
  totalAmount: number;
  notes?: string | null;
  submittedAtUtc?: string | null;
  approvedAtUtc?: string | null;
  cancelledAtUtc?: string | null;
  cancellationReason?: string | null;
  rowVersion: string;
  lines: PurchaseOrderLineDto[];
}

export interface PagedPurchaseOrdersResult {
  items: PurchaseOrderDto[];
  totalCount: number;
  page: number;
  pageSize: number;
}

export interface PurchaseOrderLineInput {
  catalogItemId?: string | null;
  descriptionSnapshot: string;
  quantity: number;
  unitCost: number;
  lineDiscountAmount?: number | null;
  sortOrder?: number | null;
}

export interface CreateDraftPurchaseOrderRequest {
  supplierId: string;
  orderDate: string;
  currencyCode: string;
  notes?: string | null;
  discountAmount?: number | null;
  lines?: PurchaseOrderLineInput[] | null;
}

File: src/Modules/Procurement/ErpClink.Modules.Procurement.Application/PurchaseOrders/Models/PurchaseOrderModels.cs
namespace ErpClink.Modules.Procurement.Application.PurchaseOrders.Models;



public sealed record PurchaseOrderLineInputDto(

    Guid? CatalogItemId,

    string DescriptionSnapshot,

    decimal Quantity,

    decimal UnitCost,

    decimal? LineDiscountAmount,

    int? SortOrder);



public sealed record CreateDraftPurchaseOrderRequest(

    Guid SupplierId,

    DateOnly OrderDate,

    string CurrencyCode,

    string? Notes,

    decimal? DiscountAmount,

    IReadOnlyList<PurchaseOrderLineInputDto>? Lines);



public sealed record UpdateDraftPurchaseOrderRequest(

    Guid SupplierId,

    DateOnly OrderDate,

    string? Notes,

    decimal? DiscountAmount,

    byte[]? RowVersion);



public sealed record AddPurchaseOrderLineRequest(

    PurchaseOrderLineInputDto Line,

    byte[]? RowVersion);



public sealed record UpdatePurchaseOrderLineRequest(

    decimal Quantity,

    decimal LineDiscountAmount,

    int SortOrder,

    byte[]? RowVersion);



public sealed record SetPurchaseOrderDiscountRequest(decimal DiscountAmount, byte[]? RowVersion);



public sealed record CancelPurchaseOrderRequest(string? Reason, byte[]? RowVersion);



public sealed record SearchPurchaseOrdersRequest(

    Guid? SupplierId,

    string? PurchaseOrderNumber,

    DateOnly? DateFrom,

    DateOnly? DateTo,

    string? Status,

    int Page = 1,

    int PageSize = 20);



public sealed record PurchaseOrderLineDto(

    Guid Id,

    Guid? CatalogItemId,

    string DescriptionSnapshot,

    decimal Quantity,

    decimal QuantityReceived,

    decimal UnitCost,

    decimal LineDiscountAmount,

    decimal LineSubtotal,

    decimal LineTotal,

    int SortOrder);



public sealed record PurchaseOrderDto(

    Guid Id,

    Guid OrganizationId,

    Guid BranchId,

    string PurchaseOrderNumber,

    Guid SupplierId,

    DateOnly OrderDate,

    string Status,

    string CurrencyCode,

    decimal SubTotal,

    decimal DiscountAmount,

    decimal TaxAmount,

    decimal TotalAmount,

    string? Notes,

    DateTime? SubmittedAtUtc,

    string? SubmittedBy,

    DateTime? ApprovedAtUtc,

    string? ApprovedBy,

    DateTime? CancelledAtUtc,

    string? CancelledBy,

    string? CancellationReason,

    DateTime CreatedAtUtc,

    string? CreatedBy,

    DateTime? UpdatedAtUtc,

    string? UpdatedBy,

    byte[] RowVersion,

    IReadOnlyList<PurchaseOrderLineDto> Lines);



public sealed record PagedPurchaseOrdersResult(

    IReadOnlyList<PurchaseOrderDto> Items,

    int TotalCount,

    int Page,

    int PageSize);

